### PR TITLE
ENDOC-545-B new branch for Run-local

### PR DESCRIPTION
**Pull request by jyunmitch**

### CODE REVIEW

File: vuepress/docs/next/tutorials/create/ms/run-local.md [hunk 1]
```diff
@@ -2,159 +2,92 @@
...
-6.  Log in using:
-
-        Username: user
-        Password: user
 
-7.  Once logged in, you will see the widget form.
+### Run the conference-form MFE
 
-#### Form Widget notes:
+1.  If you are still running the previous micro frontend, click `Ctrl+C` to end the process
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/tutorials/create/ms/run-local.md:44]: to avoid the issue of specifying / not specifying the previous mfe, especially since this applies to any mfe and not one in particular, maybe just rephrase the instruction in the context of why it's important, like you need to ensure the localhost port is available to test the current mfe?

File: vuepress/docs/next/tutorials/create/ms/run-local.md [hunk 2]
```diff
@@ -2,159 +2,92 @@
...
-1. If you are already running a widget, click `Ctrl+C` to end the process
-
-2. Go to the details widget folder in your project:
-
-        cd ui/widgets/YOUR-ENTITY-NAME/detailsWidget
-
-3. Install and start your widget with the command:
-
-        npm install && npm start
+The form to enter the name and location for conferences should open in your browser. You may enter and save new data to see it in the other MFEs.
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/tutorials/create/ms/run-local.md:50]: "of conferences" (not *for*)

something like "the data you enter and save will appear in the other MFEs" avoids the singular awkwardness and flows better and avoids the implication that changing the data in other MFEs is for its own sake, just to see the updates

File: vuepress/docs/next/tutorials/create/ms/run-local.md [hunk 3]
```diff
@@ -2,159 +2,92 @@
...
+>If you want to modify a different row in the database, edit the `microfrontends/conference-form/public/index.html` file. Change the `id` attribute in this line:
+``` html
+   <conference-form id="1" />
+```
 
-#### Widget Details notes:
+### Run the conference-details MFE
 
-If you want to modify a different row in the database, edit the `index.html` file in the public folder `ui/widgets/YOUR-ENTITY-NAME/detailsWidget/public`. Change the `id` attribute in this line:
+1. If you are running the previous micro frontend, click `Ctrl+C` to end the process
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/tutorials/create/ms/run-local.md:59]: same comment

File: vuepress/docs/next/tutorials/create/ms/run-local.md [hunk 4]
```diff
@@ -2,159 +2,92 @@
...
 
-2. By default, Docker Compose recreates the Keycloak container and resets the H2 database each time it is started. To persist Keycloak changes across restarts, you can make the following updates: 
-   * In your src/main/docker/keycloak.yml file, replace\
-   `
-   '-Dkeycloak.migration.strategy=OVERWRITE_EXISTING',` with 
-   ```
-   '-Dkeycloak.migration.strategy=IGNORE_EXISTING',
-   ```   
-   * In the same file, add a persistent volume under `volumes`:
+2. In this blueprint generated project, Docker Compose persists Keycloak data across restarts by default. If you want your data to reset on restarts:
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/tutorials/create/ms/run-local.md:77]: blueprint-generated

File: vuepress/docs/next/tutorials/create/ms/run-local.md [hunk 5]
```diff
@@ -2,159 +2,92 @@
...
+   
+   * Edit the `svc/keycloak.yml` file. Replace `-Dkeycloak.migration.strategy=IGNORE_EXISTING`,
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/tutorials/create/ms/run-local.md:79]: why is the comma included in the edit? can just say replace ignore... with overwrite... ?

File: vuepress/docs/next/tutorials/create/ms/run-local.md [hunk 6]
```diff
@@ -2,159 +2,92 @@
...
    ```
+ 
+   * In the same file, remove this persistent volume statement under `volumes`:
+   ``` yaml
    - ./keycloak-db:/opt/jboss/keycloak/standalone/data
    ```   
-   * Keycloak changes should now be persistent. You can reset your Keycloak database by emptying the `src/main/docker/keycloak-db` directory and restarting the container.
+   * Keycloak should now reset every time you restart.
+
+2. Error `User is not authenticated` : If you see this message after starting the MFEs, it is likely that your Keycloak application is not running or the `.env.local` file is not configured properly. Check if Keycloak is running; otherwise, update the configuration in `microservice/conference-x/.env.local`.
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/tutorials/create/ms/run-local.md:91]: would rephrase "if you see"; whether or not you see it is different from whether or not it's generated

File: vuepress/docs/next/tutorials/create/ms/run-local.md (hunk 1 above)
Comment (jyunmitch) [reply vuepress/docs/next/tutorials/create/ms/run-local.md:44]: I don't think it's that confusing to point out

File: vuepress/docs/next/tutorials/create/ms/run-local.md (hunk 2 above)
Comment (jyunmitch) [reply vuepress/docs/next/tutorials/create/ms/run-local.md:50]: don't think it sounds awkward

File: vuepress/docs/next/tutorials/create/ms/run-local.md (hunk 5 above)
Comment (jyunmitch) [reply vuepress/docs/next/tutorials/create/ms/run-local.md:79]: comma is part of the code, it should be inside the backtick

File: vuepress/docs/next/tutorials/create/ms/run-local.md (hunk 6 above)
Comment (jyunmitch) [reply vuepress/docs/next/tutorials/create/ms/run-local.md:91]: I think this is really down in the weeds... and a style issue

File: vuepress/docs/next/tutorials/create/ms/run-local.md [hunk 7]
```diff
@@ -18,7 +18,7 @@ The following steps utilize the ent bundle CLI.
 ``` sh
 ent bundle svc start keycloak
 ```
-2. Start up the Spring Boot application `conference-ms`. The logs will display in the console and you can stop the application with `Ctrl+C`. Keep the microservice running while the MFEs run.
+2. Start up the Spring Boot application `conference-ms`. The logs will display in the console and you can stop the application with `Ctrl+C`. Keep the microservice running while the MFEs are run.
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/tutorials/create/ms/run-local.md:21]: my vote is "MFEs" are running (common phrasing/understanding trumps the word repetition) but if you're set against that then "MFEs run" like you had before

File: vuepress/docs/next/tutorials/create/ms/run-local.md (hunk 1 above)
Comment (Lyd1aCla1r3) [reply vuepress/docs/next/tutorials/create/ms/run-local.md:44]: i guess it depends on the audience. not knowing much about how things work the user wouldn't realize that the whole point is to free up the port. i kind of think it's worth noting, just tacking that on the end of the sentence for clarity, but your call

File: vuepress/docs/next/tutorials/create/ms/run-local.md (hunk 2 above)
Comment (Lyd1aCla1r3) [reply vuepress/docs/next/tutorials/create/ms/run-local.md:50]: it's interpretation, not style

File: vuepress/docs/next/tutorials/create/ms/run-local.md (hunk 6 above)
Comment (Lyd1aCla1r3) [reply vuepress/docs/next/tutorials/create/ms/run-local.md:91]: kind of. more about precision

File: vuepress/docs/next/tutorials/create/ms/run-local.md (hunk 7 above)
Comment (jyunmitch) [reply vuepress/docs/next/tutorials/create/ms/run-local.md:21]: the run here is used in the command sense of the word, as in run the command. The tutorial is about running the components. I think this is fine and more accurate.

Comment (Lyd1aCla1r3) [reply vuepress/docs/next/tutorials/create/ms/run-local.md:21]: i can appreciate that, but i think it's either "while the MFEs are running" or "when the MFEs are run"... "while" implies continued duration, whereas executing a command is done in an instant

**Review by Lyd1aCla1r3 — APPROVED** [review]
